src/Raven.Server/Documents/AI/Embeddings/QueryEmbeddingsBatchingWorker.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Embeddings;
+using Raven.Server.Background;
+using Raven.Server.Config;
+using Raven.Server.Config.Categories;
+using Raven.Server.Documents.ETL.Providers.AI;
+using Raven.Server.Documents.ETL.Providers.AI.Embeddings;
+using Sparrow.Server.Logging;
+
+namespace Raven.Server.Documents.AI.Embeddings
+{
+    public class QueryEmbeddingsBatchingWorker(string databaseName,
```

Comment:
I really don't like this implementation - it is quite complex, and run into output issues, threading, etc.
The model should be:

- N tasks, all waiting on an event
- Query comes in and deposit text to be embedded, raise the event
- One of the tasks wakes up, take the lock and then grab as much as it can beom the queue
- Start async embedding generation
- Release the lock
- If there is any activity, the next taks runs and take as much as it can, etc...